### ISSUE TITLE
Add a "Keep Both" strategy?

### ISSUE THREAD

**Issue by al-the-x** [smashwilson/merge-conflicts#185]
The "Keep Left" / "Keep Right" option is occasionally useful; often I need a combination of Left and Right. Most graphically diff-ing tools support a "Keep Both" option that allows me to slot the Left changes ahead or behind the Right changes. Would you consider adding this feature?

**Comment by smashwilson** [comment]
It already exists :grin:

Right-click anywhere in the conflict and you can choose to resolve "ours then theirs" or "theirs then ours" from the context menu.

**Comment by al-the-x** [comment]
Ah, good to know! It wasn't immediately obvious, though... Maybe an auxiliary button in the UI that indicates that other actions are available would be nice? Happy to submit some more concrete ideas (pics and code), if you're interested...

**Comment by philipgiuliani** [comment]
I also didn't knew about this option and missed it so often. Cool that it exists. Maybe it should be a button. Its not so obvious that you have to right-click in the conflict :/

**Comment by smashwilson** [comment]
Yah, I've been thinking about revisiting the conflict-marker UI a little, and this is one of the considerations. Maybe a gear that exposes the menu for "other resolutions"?

**Comment by al-the-x** [comment]
:+1:

**Comment by al-the-x** [comment]
Interested in a PR for this? Point me in the right direction...

**Comment by al-the-x** [comment]
Ping @smashwilson: Any progress on a UI improvement for "Keep Both" as suggested in this thread?

**Comment by smashwilson** [comment]
Nope, sorry, I haven't had chance to allocate time to this for a while now (day job, you know).

**Comment by al-the-x** [comment]
Totally. Point me in the right direction for a PR?

**Comment by smashwilson** [comment]
Well, the view that owns the controls is [SideView](https://github.com/smashwilson/merge-conflicts/blob/master/lib/view/side-view.coffee#L7-L13); I've highlighted the markup generation. The resolution commands are defined and implemented in [ConflictedEditor](https://github.com/smashwilson/merge-conflicts/blob/master/lib/conflicted-editor.coffee#L74-L81). That should get you started, I think?

**Comment by al-the-x** [comment]
Thanks, yes. That demystifies a lot. I haven't done any work on an Atom plugin yet, but I'm eager to learn. I'll fork and open a PR this week.

**Comment by smashwilson** [comment]
:wave: Thanks for taking the time to file an issue here! Unfortunately, this repository is being [deprecated](https://github.com/smashwilson/merge-conflicts/blob/master/docs/announcement.md), so I won't be getting to this one here.

This one should be taken care of by the [new git and GitHub package for Atom](https://github.com/atom/github). We have "ours then theirs" and "theirs then ours" options in a drop-down menu on the conflict itself, so hopefully it'll be a bit more discoverable.
